tensorflow/python/keras/utils/vis_utils.py

```diff
@@ -124,6 +146,27 @@ def model_to_dot(model,
     dot.set('dpi', dpi)
     dot.set_node_defaults(shape='record')
 
+  if layer_range is None:
+    layer_range = [
+      model.get_layer(index=0).name,
+      model.get_layer(index=len(model.layers)-1).name
+    ]
+  
+  # check point for layer range whether its string or integer 
+  # layer range must be same dtype (mixed dtype is not allowed)
+  if len(set([type(x) for x in layer_range])) > 1:
```

Comment:
You can just check if the layer_range[0] and layer_range[1] are both str type.

Reply:
done